test/syntax-interpreter.spec.js

```diff
@@ -220,29 +220,42 @@ describe('SyntaxInterpreter', () => {
 
     it('handles a semicolon inside a string of an options attribute', () => {
       info.attrValue = "foo: 'bar;';";
-      var instruction = interpreter.options({}, null, info, null);
+      let instruction = interpreter.options({}, null, info, null);
       expect(instruction.attributes['foo']).toBe("'bar;'");
     });
 
     it('handles an escaped single quote inside a string of an options attribute', () => {
       info.attrValue = "foo: 'bar\\'';";
-      var instruction = interpreter.options({}, null, info, null);
+      let instruction = interpreter.options({}, null, info, null);
       expect(instruction.attributes['foo']).toBe("'bar\\''");
     });
 
     it('handles an escaped single quote and a semicolon inside a string of an options attribute', () => {
       info.attrValue = "foo: 'bar\\';';";
-      var instruction = interpreter.options({}, null, info, null);
+      let instruction = interpreter.options({}, null, info, null);
       expect(instruction.attributes['foo']).toBe("'bar\\';'");
     });
 
     it('handles multiple properties', () => {
       info.attrValue = "foo: 'bar;'; abc.bind: xyz; hello: ${world}; optimus: ${prime ? 'decepticon;' : ';'} test";
-      var instruction = interpreter.options({}, null, info, null);
+      let instruction = interpreter.options({}, null, info, null);
       expect(instruction.attributes['foo']).toBe("'bar;'");
       expect(instruction.attributes['abc.bind']).toBe("xyz");
       expect(instruction.attributes['hello']).toBe("${world}");
       expect(instruction.attributes['optimus']).toBe("${prime ? 'decepticon;' : ';'} test");
     });
+
+    it('handles single unnamed option with default property', () => {
+      let resources = { getAttribute(name) {} };
+
+      info.attrValue = "bar";
+
+      spyOn(resources,'getAttribute').and.returnValue({
+        defaultProperty: { name: 'foo' }
+      });
+
+      let instruction = interpreter.options(resources, null, info, null, { attributeName: 'foo' });
+      expect(instruction.attributes['foo']).toBe('bar');
+    });
   });
```

Comment:
I added a bunch of tests with and without semilcolons, and with and without the name of the property.  All the tests succeed with the existing code.  However, your suggested change is more efficient and clear, so it makes sense to make the change in any case.

However, one test I tried fails, with either version of the code, namely this one:  "bar;"

Is this legal syntax?  What should be the expected result in this case?  Do you want to tolerate it?